src/hotspot/share/opto/c2_globals.hpp

```diff
@@ -806,6 +806,10 @@
   product(bool, UseStoreStoreForCtor, true, DIAGNOSTIC,                     \
           "Use StoreStore barrier instead of Release barrier at the end "   \
           "of constructors")                                                \
+                                                                            \
+  product(uintx, ShortLoopIter, 1000,                                       \
```

Comment:
May I ask why this is such a small value, why is it not a value that is closer to the limit of an `int` loop, such as 10^6?

Reply:
1000 happens to be the loop strip mining limit too. So by restricting the number of iterations to 1000, we also get rid of the outer loop for loop strip mining.
It could make sense to have a similar mechanism for loop strip mining and then have long counted loops and loop strip mined loops have their own tests for what's a small number of iterations.